.github/workflows/publish-release.yml

```diff
@@ -27,3 +27,33 @@ jobs:
       - uses: MetaMask/action-publish-release@v1
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - run: |
+          yarn setup
+          yarn build
+      - uses: actions/cache@v2
+        id: restore-build
+        with:
+          path: ./*
+          key: ${{ github.sha }}
+
+  publish-npm:
+    environment: publish
```

Comment:
we'll need to create this `environment` with whatever restrictions we want for publish. we can require reviews from a team of developers and the workflow will not run until approval is met.

Reply:
That sounds like a good idea. How would we do that?

Reply:
Ah, just found the link you posted in the description. Would we have to set up a `publish` environment per-repo?

Reply:
> Would we have to set up a publish environment for every repo...

Yes I believe that's the case now, but there's potentially a future where a single environment could be setup at the org level and used in multiple repositories (I'm assuming).